src/content/docs/en/guides/migrate-to-astro/from-create-react-app.mdx

```diff
@@ -0,0 +1,379 @@
+
+### Add your existing app files
+
+Copy your existing Create React App (CRA) project source files and folders (e.g. `components`, `hooks`, `styles`, etc.) into a new folder inside `src/`, keeping its file structure so your app will continue to work. Note that all `.js` file extensions must be renamed to `.jsx` or `.tsx`.
+
+Do not include any configuration files. You will use Astro's own `astro.config.mjs`, `package.json`, and `tsconfig.json`.
+
+Move the contents of your app's `public/` folder (e.g. static assets) into Astro's `public/` folder.
+
+<FileTree>
+- public/
+  - logo.png
+  - favicon.ico
+  - ...
+- src/
+  - cra-project/
+    - App.jsx
+    - ...
+  - pages/
+    - index.astro
+- astro.config.mjs
+- package.json
+- tsconfig.json
+</FileTree>
+
+### Render your app
+
+Import your app's root component in the frontmatter section of `index.astro`, then render the `<App />` component in your page template:
+
+```astro title="src/pages/index.astro"
+---
+import App from '../cra-project/App.jsx'
+---
+<App client:load/>
+```
+
+:::note[client directives]
+Your app needs a [client directive](/en/reference/directives-reference/#client-directives) for interactivity. Astro will render your React app as static HTML until you opt-in to client-side JavaScript.
+
+Use `client:load` to ensure your app loads immediately from the server, or `client:only="react"` to skip server-rendering and run your app entirely client-side.
+:::
+
+### Check the Astro config file
+
+Astro has a configuration file at the root of your project called [`astro.config.mjs`](/en/guides/configuring-astro/). This is used for configuring your Astro project and any installed integrations, including [SSR adapters](/en/guides/deploy/).
+
+You should see the React integration installed, along with any other official Astro integrations you added.
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config'
+
+export default defineConfig({
+  integrations: {
+    [react()]
+  }
+  // your configuration options here...
+  // https://docs.astro.build/en/reference/configuration-reference/
+})
+```
+
+See our guide for [configuring Astro](/en/guides/configuring-astro/) for more details and available options.
+
+## Convert your CRA to Astro
+
+After [adding your existing app to Astro](#add-your-create-react-app-to-astro), you will probably want to convert your app itself to Astro! 
+
+You will replicate a similar component-based design [using Astro HTML templating components for your basic structure](/core-concepts/astro-components/) while importing and including individual React components (which may themselves be entire apps!) for islands of interactivity.
+
+Every migration will look different, and can be done incrementally without disrupting your working app. Convert individual pieces at your own pace so that more and more of your app is powered by Astro components over time. 
```

Comment:
```suggestion
Every migration will look different and can be done incrementally without disrupting your working app. Convert individual pieces at your own pace so that more and more of your app is powered by Astro components over time.
```